kafka-connect-oura-source/src/main/java/org/radarbase/connect/rest/oura/OuraRestSourceConnectorConfig.java

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright 2018 The Hyve
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.radarbase.connect.rest.oura;
+
+import static org.apache.kafka.common.config.ConfigDef.NO_DEFAULT_VALUE;
+
+import java.lang.reflect.InvocationTargetException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import okhttp3.Headers;
+import okhttp3.HttpUrl;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.NonEmptyString;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.Validator;
+import org.apache.kafka.common.config.ConfigDef.Width;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.storage.OffsetStorageReader;
+import org.radarbase.connect.rest.config.ValidClass;
+import org.radarbase.connect.rest.oura.request.OuraReqGenerator;
+import org.radarbase.connect.rest.oura.user.OuraServiceUserRepository;
+
+public class OuraRestSourceConnectorConfig extends AbstractConfig {
+  public static final Pattern COLON_PATTERN = Pattern.compile(":");
+
+  private static final String SOURCE_POLL_INTERVAL_CONFIG = "rest.source.poll.interval.ms";
+  private static final String SOURCE_POLL_INTERVAL_DOC = "How often to poll the source URL.";
+  private static final String SOURCE_POLL_INTERVAL_DISPLAY = "Polling interval";
+  private static final Long SOURCE_POLL_INTERVAL_DEFAULT = 60000L;
+
+  static final String SOURCE_URL_CONFIG = "rest.source.base.url";
+  private static final String SOURCE_URL_DOC = "Base URL for REST source connector.";
+  private static final String SOURCE_URL_DISPLAY = "Base URL for REST source connector.";
+
+  private static final String SOURCE_REQUEST_GENERATOR_CONFIG = "rest.source.request.generator.class";
```

Comment:
@yatharthranjan Do we need to specify this in the config since there's only one request generator?